### DISCUSSION TITLE
Trip to Klipper.  But quickly back to Professional Firmware, and this time with aml!

### DISCUSSION THREAD

**Discussion by heidepiek** [mriscoc/Ender3V2S1#1288] [Show and tell]
![benchy](https://github.com/mriscoc/Ender3V2S1/assets/104028471/ea00bc40-1dbb-4594-a03f-f5c8fa29d2ce)

If I am correct (years go by so fast), I have been using the professional firmware since the end of 2021. Always with pleasure. Many features that creality always deprived you of. Improved a lot over the years. A while ago i bought a k1 max. After some struggles rooted (couple of bright minds who already figured all that out) the printer. After that very nice results. So I conceived the plan to install klipper on 1 of my ender 3v2`s. After a few weeks of tweaking, finally lit up the neopixels and got my powerplug with tasmota going with klipper. But the print results weren't exactly to be happy about. Lots of hours later, belts tightening loosening. Extruder calibration, pid tuning. rollers tuning. I just couldn't get the printer to print properly. The firmware finally installed properly. But no good print results. Well, I was not amused. So I quickly made a new bin of the familiar professional firmware. Flashed it on the printer, tweaked settings. And voilla, a neat benchy. Miguel, in the meanwhile also introduced aml. Thought, let me try that too. It cost me 12.50 (money well spent and miguel deserves it too! a lot of hours programming in this firmware) After reading the attached pdf again!!!!  (my printer did a home, went into wait position. warmed up, but stuck in there) I read the bit where it says install Pillow. After installing, voilla: a perfectly working aml!
A tip for the people who like to use this professional firmware, read the wiki! If you have problems, usually the solution is in the wiki. Otherwise, there is a search bar above the discussions section. A lot of things are listed here, and often solved. Lots of bright minds working together to solve problems. Miguel, thanks for this great firmware! Know that a lot more people are working on this, kudos to them as well!!!!  For the rest of you, happy printing. 

Greetz
Harrie

![benchy1](https://github.com/mriscoc/Ender3V2S1/assets/104028471/76a33ec8-cd94-4f78-a696-b6aeaad502e5)

https://github.com/mriscoc/Ender3V2S1/assets/104028471/76a6e5d9-0203-42a8-a0a7-ba9ac892187e

**Comment by mriscoc** [comment]
Nice work and printer, congratulations and thanks for the AML purchase.
